pkg/innerring/config.go

```diff
@@ -181,10 +181,34 @@ func parseBlockchainConfig(v *viper.Viper, _logger *logger.Logger) (c blockchain
 
 	const rpcSection = rootSection + ".rpc"
 	if v.IsSet(rpcSection) {
-		c.RPCListenAddresses, err = parseConfigAddressesTCP(v, rpcSection+".listen", "network addresses to listen Neo RPC on")
+		c.RPC.Addresses, err = parseConfigAddressesTCP(v, rpcSection+".listen", "network addresses to listen Neo RPC on")
 		if err != nil && !errors.Is(err, errMissingConfig) {
 			return c, err
 		}
+
+		const rpcTLSSection = rpcSection + ".tls"
+		if v.GetBool(rpcTLSSection + ".enabled") {
+			c.RPC.TLSConfig.Enabled = true
```

Comment:
pretty useless, `c.RPC.TLSConfig.Addresses` is required to be non-empty and that's enough to determine the behavior

Reply:
> pretty useless, `c.RPC.TLSConfig.Addresses` is required to be non-empty

why?

Reply:
cause IIUC if `enabled` flag is set then `addresses` are required, isn't it?

Reply:
> cause IIUC if enabled flag is set then addresses are required, isn't it?

i would say "no". while sound useless, it is not as strange as `tls.enabled` but no `tls.cert_file` for me (which is required with the last push). it is enabled but listens to no addresses. like an empty `for` loop. why not?

Reply:
ok lets stick to proposed approach, will change later if needed